mission_control/include/mission_control/mission.h

```diff
@@ -65,58 +64,31 @@ class Mission
     PAUSED,
     COMPLETE
   };
-  void SetState(MissionState state);
-  MissionState GetState();
-  boost::mutex m_MissionStateLock;
-
-  void ExecuteMission(ros::NodeHandle nh);
-  void AbortMissionWrapper(ros::NodeHandle nh);
 
+  void processMission();
+  void ExecuteMission();
   void Stop();
 
-  boost::shared_ptr<boost::thread> m_threadProcMission, m_threadAbortMission;
-
-  void ProcessCorrectedPoseData(const pose_estimator::CorrectedData &data);
+  int getMissionStatus();
 
-  void setMissionDescription(std::string descStr) { m_mission_description = descStr; }
-  std::string getMissionDescription() { return m_mission_description; }
+  std::string getCurrenMissionDescription();
+  std::string getCurrentBehavioralName();
 
-  std::string getCurrentBehaviorsName() { return current_behavior_name; }
-  std::string getCurrentAbortBehaviorsName() { return current_abort_behavior_name; }
-
-  int getCurrentBehaviorId() { return current_behavior_id; }
-  int getCurrentAbortBehaviorId() { return current_abort_behavior_id; }
+  bool loadBehavior(const std::string &missionFullPath);
+  
+  int getCurrentBehaviorID();
+  
+  BT::Tree _missionTree;
+  NodeStatus _missionStatus;
```

Comment:
@patapufate why are these member fields public?

Reply:
Forgot to move to the private section.